sdk/schema/plugin.go

```diff
@@ -117,3 +118,23 @@ func (p Plugin) DeepValidate() []ValidationReport {
 
 	return reports
 }
+
+func (p Plugin) MarshalJSON() ([]byte, error) {
+	if len(p.Credentials) == 0 {
+		return nil, nil
+	}
+	plugin := struct {
+		PlatformName  string `json:"platform_name"`
+		ManagementURL string `json:"management_url"`
```

Comment:
We can consider making this `omitempty`. 

However, this would make things a bit harder in the closed-source code, so I propose we leave it like it is for now.